lib/ensure_content_type.js

```diff
@@ -33,12 +33,16 @@ function checkContentType(hyper, req, next, expectedContentType, responsePromise
 
         function updateSpecWarning() {
             // Returned content type version is higher than what the
-            // spec promises. Log a warning about the need to update the spec.
-            hyper.log('warn/content-type/spec_outdated', {
-                msg: 'Spec needs update to reflect latest content type.',
-                expected: expectedContentType,
-                actual: res.headers['content-type']
-            });
+            // spec promises.
+            if (Math.random() < 0.001) {
+                // Log a warning about the need to update the spec. Sampled 1:1000.
+                hyper.log('warn/content-type/spec_outdated', {
```

Comment:
An outdated spec is not normal & requires attention, so I think a warning is the best way to ensure that users interested in warnings see this. 

Sampled_levels on the other hand is an opt-in mechanism that lets us log specific items from lower-priority levels, which would not otherwise be logged. We shouldn't rely on users setting that up for specific messages just to see warnings.